Refs #28723 -- Fixed stale prefetch_related cache after add/remove.

jdufresne opened the pull request:
Regression in 514b2c989a948e3c59bda0da0c9427acf643cf5b.

timgraham commented on tests/prefetch_related/test_prefetch_related_objects.py:
```diff
@@ -32,6 +34,34 @@ def setUpTestData(cls):
+        new_book = Book.objects.create(title='Ulysses')
+        prefetch_related_objects([p], 'book_set')
+        self.assertEqual(
+            set(p.book_set.all()),
```
Is the set() stuff to avoid ordering differences? Usually we use `assertCountEqual` for that. (I guess you copied existing tests.)

timgraham commented on tests/prefetch_related/test_prefetch_related_objects.py:
```diff
@@ -32,6 +34,34 @@ def setUpTestData(cls):
         cls.reader1.books_read.add(cls.book1, cls.book4)
         cls.reader2.books_read.add(cls.book2, cls.book4)
 
+    def test_add_clears_prefetched_objects(self):
```
Considering the docstring of this test class, I would put the test in `tests.py`.

jdufresne commented:
Thanks for the review. Updated the commit:

- Moved tests to `tests.py`. Now in same test class from the regression commit. Let me know if you think there is a better one.
- Change assert to `assertCountEqua()`. (Made new PR to fix existings tests #9661.)
- Use existing models instead of creating a new one.
